tiledb/sm/subarray/subarray.cc

```diff
@@ -2968,13 +2968,8 @@ Status Subarray::compute_relevant_fragments(
   }));
 
   // Recalculate relevant fragments.
-  if (!relevant_fragments_.has_value()) {
-    relevant_fragments_ = std::vector<unsigned>();
-  } else {
-    relevant_fragments_->clear();
-  }
-
-  relevant_fragments_->reserve(fragment_num);
+  relevant_fragments_.clear_computed_relevant_fragments();
```

Comment:
Inline mutation like this is one of the things we need to work towards getting rid of; it makes reasoning about the state of both objects much more difficult.

It looks like there is substantial overlap between `RelevantFragments` and `ComputeRelevantFragmentsCtx` (we're looping over one below in order to create the other). Can we merge those classes? Or, if not, add a method to `RelevantFragments` which initializes from a fragment bytemap vector (or updates, if absolutely necessary w/ the current code structure).